storage2/src/state.rs

```diff
@@ -0,0 +1,47 @@
+use std::collections::HashMap;
+
+use jmt::{
+    storage::{TreeReader, TreeWriter},
+    JellyfishMerkleTree,
+};
+
+mod transaction;
+use transaction::Transaction as StateTransaction;
+
+/// State is a lightweight copy-on-write fork of the chain state,
+/// implemented as a RYW cache over a pinned JMT version.
+pub(crate) struct State<'a, R: TreeReader + TreeWriter> {
+    cache: HashMap<jmt::KeyHash, jmt::OwnedValue>,
+    jmt_version: jmt::Version,
+    jmt: &'a JellyfishMerkleTree<'a, R>,
```

Comment:
We can't know what data the `State` will need to maintain until we design how it works in #1490, so we should not fill it in with fields that might be wrong.